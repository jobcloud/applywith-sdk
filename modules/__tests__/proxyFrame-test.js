/**
 * @jest-environment jsdom
 * @flow
 */
import render, { reset } from '../proxyFrame';

const createFrame = path => {
  const container = document.createElement('div');
  render(container, path);
  return container;
};

describe('proxyFrame', () => {
  afterEach(() => {
    reset();
  });
  it('renders anchor in container', () => {
    const container = createFrame();
    expect(container.children.length).toBe(1);
    expect(container.children[0].tagName).toBe('IFRAME');
  });
  it('renders frame with correct default structure', () => {
    const anchor = createFrame();
    expect(anchor.innerHTML).toMatchSnapshot();
  });
  it('renders frame with overwritten pathname', () => {
    const origin = 'http://localhost:8081';
    const pathname = `${origin}/button.html`;
    const frame = createFrame(pathname).children[0];
    expect(frame.getAttribute('src')).toContain(pathname);
  });
});
