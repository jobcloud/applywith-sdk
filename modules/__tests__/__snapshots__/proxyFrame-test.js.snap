// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proxyFrame renders frame with correct default structure 1`] = `
"<iframe src=\\"https://www.jobs.ch/public/support/oauth-xdomain-proxy.html\\" frameborder=\\"0\\" width=\\"0\\" height=\\"0\\" id=\\"jobcloud-xdomain-frame\\" style=\\"
  visibility: hidden;
\\"></iframe>"
`;
